File: editor/en/[2997]Minimum Number of Operations to Make Array XOR Equal to K.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int minOperations(int[] nums, int k) {
        int xorSum = 0;

        for (int num : nums) {
            xorSum ^= num;
        }

        int difBits = xorSum ^ k;
        return Integer.bitCount(difBits);
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2997]Minimum Number of Operations to Make Array XOR Equal to K.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        xor_sum = 0;

        for num in nums:
            xor_sum ^= num

        dif_bits = xor_sum ^ k
        return bin(dif_bits).count('1');
# leetcode submit region end(Prohibit modification and deletion)
